src/adapter/src/coord/peek.rs

```diff
@@ -423,6 +493,102 @@ impl crate::coord::Coordinator {
+                            let Some(count) = NonZeroUsize::new(count) else {
+                                continue;
+                            };
+                            let mut datum_local = datum_vec.borrow_with(&row);
+                            let eval_result = mfp_plan
+                                .evaluate_into(&mut datum_local, &arena, &mut row_builder)
+                                .map_err(|e| e.to_string())?;
+                            if let Some(row) = eval_result {
+                                accum.push((row, count));
+                                limit_remaining = limit_remaining.saturating_sub(count.get());
+                                if limit_remaining == 0 {
+                                    break;
+                                }
+                            }
+                        }
+                    }
+
+                    let res = finishing.finish(accum, max_result_size)?;
+                    let total_duration = start.elapsed();
+                    metrics
+                        .persist_peek_seconds
+                        .observe(total_duration.as_secs_f64());
+                    info!(
+                        collection =? id,
+                        fetch_duration =? last_fetch,
+                        total_duration =? total_duration,
+                        "persist peek success"
+                    );
+                    Ok(PeekResponseUnary::Rows(res))
+                });
+            let internal_cmd_tx = self.internal_cmd_tx.clone();
+            let ctx_extra = mem::take(ctx_extra);
+            return Ok(ExecuteResponse::SendingRows {
+                future: Box::pin(async move {
+                    let res = match handle.await {
+                        Ok(Ok(res)) => res,
+                        Ok(Err(err)) => PeekResponseUnary::Error(err.to_string()),
+                        Err(_join_error) => PeekResponseUnary::Canceled,
+                    };
+
+                    let reason = match &res {
+                        PeekResponseUnary::Rows(rows) => StatementEndedExecutionReason::Success {
+                            rows_returned: Some(u64::cast_from(rows.len())),
+                            execution_strategy: Some(StatementExecutionStrategy::FastPath),
+                        },
+                        PeekResponseUnary::Error(e) => {
+                            StatementEndedExecutionReason::Errored { error: e.clone() }
+                        }
+                        PeekResponseUnary::Canceled => StatementEndedExecutionReason::Canceled,
+                    };
+
+                    let tx_send = internal_cmd_tx.send(Message::RetireExecute {
+                        data: ctx_extra,
+                        reason,
+                    });
+                    if tx_send.is_err() {
+                        return PeekResponseUnary::Canceled;
+                    }
```

Comment:
We generally ignore these errors and log a warning. They generally only happen when the Coordinator thread has shut down before this task. For example:

https://github.com/MaterializeInc/materialize/blob/b5e366cea8acb94e95b808db2ab3918cfb77cfc2/src/adapter/src/coord.rs#L879-L884